### ISSUE TITLE
tensorflow label_image recognize so slow

### ISSUE THREAD

**Issue by shaolinkhoa** [tensorflow/tensorflow#5785]
I already asked [this question](http://stackoverflow.com/questions/40705203/tensorflow-label-image-recognize-so-slow?noredirect=1#comment68687966_40705203) on stackoverflow but It seems no one knows.

would you mind giving advices to make to code run faster.
Cause It took 5 second on laptop (30 seconds on pi 2) to recognize one picture of letter  

```
(tensorflow)khoa@khoa:~/tensorflow/tf_lv$ time python label_image.py photo1/9_9.jpg 
W tensorflow/core/framework/op_def_util.cc:332] Op BatchNormWithGlobalNormalization is deprecated. It will cease to work in GraphDef version 9. Use tf.nn.batch_normalization().
t

real	0m2.344s
user	0m3.144s
sys	0m0.512s
```

**Comment by prb12** [comment]
Please can you provide all of the information requested in the issues reporting template.

**Comment by shaolinkhoa** [comment]
I already asked [this question](http://stackoverflow.com/questions/40705203/tensorflow-label-image-recognize-so-slow?noredirect=1#comment68687966_40705203) on stackoverflow but It seems no one knows.

**Operating System:** Ubuntu 14.04 LTS

**Installed version of tensorflow:**   0.10.0

My project is OCR. I used **image_retraining**(v0.10.0) to recognize letters.

I train it with pictures size 128x128

After that I use my code to input several letter pictures (1306 pictures) which I segmented from a page of document

The code run so slow.

    It took 3 seconds to recognize 1 letter and near 30 minutes to finish 1306 pictures on my laptop.

    It took 38 seconds to recognize 1 letter and near 6 hours to finish 1306 pictures on pi 2

I don't know why it run so slow. My C++ code use SVM on QT just took 5 seconds to do that ( It uses picture size 32x24).

So I think It's because I use picture too large or python run slower than C++ 

Update #1: The picture size is not the problem. I tried with the small picture size 20x20 but it's still slow. 
Follow the [time_chart](https://github.com/shaolinkhoa/tensorflow/blob/master/time_chart.txt). 
```
  ncalls  tottime  percall  cumtime  percall filename:lineno(function)
     1309    0.029    0.000  990.377    0.757 /home/khoa/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py:599(run)
     1309    0.134    0.000  990.333    0.757 /home/khoa/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py:835(_run)
     1309    0.010    0.000  989.609    0.756 /home/khoa/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py:916(_do_run)
     1309    0.004    0.000  989.599    0.756 /home/khoa/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py:963(_do_call)
     1309    0.040    0.000  989.595    0.756 /home/khoa/tensorflow/local/lib/python2.7/site-packages/tensorflow/python/client/session.py:940(_run_fn)
    1309 988.335    0.755  988.335    0.755 {_pywrap_tensorflow.TF_Run}
     2646     56.390 0.021  56.390      0.021 {posix.system}
```

It seems the code slow because of this command

predictions = sess.run(softmax_tensor, \
        {'DecodeJpeg/contents:0': image_data})

Would you mind giving me advices to make it run faster

**Comment by prb12** [comment]
> I already asked this question on stackoverflow but It seems no one knows.

I'm sorry, but this is not an appropriate venue for general support questions.

From https://www.tensorflow.org/versions/r0.11/resources/index.html
> For help and support, technical or algorithmic questions, please submit your questions to Stack Overflow: https://stackoverflow.com/questions/tagged/tensorflow. You may also find answers in our FAQ, our glossary, or in the shapes, sizes and types guide. Please do not use the mailing list or issue tracker for support.

In general, for any performance related issue I would suggest that you consider capturing a Timeline as described here #3009.  If that shows that there is a specific TensorFlow op which is running slower than expected the please feel free to open a specific issue.  

I would also suggest running something like `htop` or `vmstat` to check that you do not have any virtual memory issues.

Other things I would suggest is to measure how long you spend doing I/O (to read the model parameters and construct the graph, and the time spent reading the JPEG image data).  You will also incur some substantial overheads feeding the image data and fetching the results from Python.

**Comment by Ahmed-Masoud** [comment]
did you find a solution?

**Comment by nanshihui** [comment]
any volunteer? 3s a picture. tooo slow
